components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -413,9 +408,6 @@ private PostAuthnHandlerFlowStatus handleRequestFlow(HttpServletRequest request,
                                     localClaimValues.get(EMAIL_ADDRESS_CLAIM))) {
                         username = UserCoreUtil.addTenantDomainToEntry(username, context.getTenantDomain());
                     }
-                    if (StringUtils.isEmpty(associatedLocalUser)) {
-                        isUsernameExists(context, username);
```

Comment:
What will happen if a customer uses custom jsp file? won't it cause an issue as we are skipping a isUsernameExists function?

Reply:
To identify if the user is existing, same as the defualt implementation, the user will have to first performa a username validation using the `api/identity/user/v1.0/validate-username` endpoint. This will throw an error if the username is already existing. 
If the custom jsp skips this check, the federation flow will continue without the user being provisioned. However the user will not be aware of the provisioning failure unless username is validated initially and the existence of the username is handled appropriately.